feat(entity/schema): add typeHidden option

GeertWille opened the pull request:
In my projects I wanted a way to hide the returned entity type to not mess up my domain models. Happy to rename the option if anyone has a better name for it 😄

naorpeled commented:
What about using projections?

GeertWille commented:
I don't use the type anywhere in the project so adding projections for every query would be cumbersome. That's why I opted for an entity level option.

naorpeled reviewed (approved):
Overall looks good!

Let's also add a test where we have an entity type alias and also typeHidden

naorpeled commented:
Hey @GeertWille,
I've left some comments, once they're addressed I will merge this PR.

Let me know if you have any questions 🙏

naorpeled commented:
> > Hey @GeertWille, I've left some comments, once they're addressed I will merge this PR.
> > Let me know if you have any questions 🙏
> 
> I added an additional test and clarified the comment. Let me know if anything else needs to be done. 👍🏼

imo we should remove that comment, it feels unecessary, wdyt?

GeertWille commented:
> 

Sure. Removed it!

GeertWille commented:
> > > Hey @GeertWille, I've left some comments, once they're addressed I will merge this PR.
> > > Let me know if you have any questions 🙏
> > 
> > 
> > I added an additional test and clarified the comment. Let me know if anything else needs to be done. 👍🏼
> 
> imo we should remove that comment, it feels unecessary, wdyt?

Sure. Removed it!

naorpeled commented:
> > > > Hey @GeertWille, I've left some comments, once they're addressed I will merge this PR.
> > > > Let me know if you have any questions 🙏
> > > 
> > > 
> > > I added an additional test and clarified the comment. Let me know if anything else needs to be done. 👍🏼
> > 
> > 
> > imo we should remove that comment, it feels unecessary, wdyt?
> 
> Sure. Removed it!

 I also meant the `TOIMPROVE` part, sorry for not being clear 🙏

GeertWille commented:
> `TOIMPROVE`

Ah sorry. My bad 😄 . Thanks for resolving that!
